### DISCUSSION TITLE
How does addGeneIntegration matrix impute the full transcriptome?

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#1001] [Questions / Documentation]
Hi @rcorces and @jgranja24 ,

How does ArchR impute the full transcriptome (all genes) if the nGenes parameter is set to 2000? In Seurat, this parameter returns the number of genes in the integrated expression matrix. The output of PeaktoGene Links gives all 20k genes, so I was wondering how the expression information is imputed for the remaining genes since here we only specify to use 2000 genes. 

If you could point in the direction of where this is in the source code, that would be greatly appreciated!

Thanks!

```
addGeneIntegrationMatrix(
  ArchRProj = NULL,
  useMatrix = "GeneScoreMatrix",
  matrixName = "GeneIntegrationMatrix",
  reducedDims = "IterativeLSI",
  seRNA = NULL,
  groupATAC = NULL,
  groupRNA = NULL,
  groupList = NULL,
  sampleCellsATAC = 10000,
  sampleCellsRNA = 10000,
  embeddingATAC = NULL,
  embeddingRNA = NULL,
  dimsToUse = 1:30,
  scaleDims = NULL,
  corCutOff = 0.75,
  plotUMAP = TRUE,
  UMAPParams = list(n_neighbors = 40, min_dist = 0.4, metric = "cosine", verbose =
    FALSE),
  nGenes = 2000,
  useImputation = TRUE,
  reduction = "cca",
  addToArrow = TRUE,
  scaleTo = 10000,
  genesUse = NULL,
  nameCell = "predictedCell",
  nameGroup = "predictedGroup",
  nameScore = "predictedScore",
  transferParams = list(),
  threads = getArchRThreads(),
  verbose = TRUE,
  force = FALSE,
  logFile = createLogFile("addGeneIntegrationMatrix"),
  ...
)

```

**Comment by rcorces** [comment]
I'm not sure that interpretation is correct? `nGenes` is the number of variable features to use for integration to find the transfer anchors. But once integration is performed, all expression data from the given cell is carried forward.
